core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -305,9 +305,12 @@ public List<TableIdentifier> listTables(Namespace namespace) {
     }
 
     return fetch(
-        row ->
-            JdbcUtil.stringToTableIdentifier(
-                row.getString(JdbcUtil.TABLE_NAMESPACE), row.getString(JdbcUtil.TABLE_NAME)),
+        row -> {
+          String ns = row.getString(JdbcUtil.TABLE_NAMESPACE);
```

Comment:
previously this would return the identifier as `.tblName` instead of `tblName` when using an empty namespace